core/src/main/java/com/orientechnologies/orient/core/sql/query/OSQLSynchQuery.java

```diff
@@ -78,8 +78,13 @@ public void end() {
   public List<T> run(final Object... iArgs) {
     String currentThreadName = Thread.currentThread().getName();
     try {
-      Thread.currentThread().setName(currentThreadName + " <query>" + this.getText() + "</query>");
-      
+      if (currentThreadName == null || !(currentThreadName.contains("<query>") || currentThreadName.contains("<command>"))) {
+        try {
+          Thread.currentThread().setName(currentThreadName + " <query>" + this.getText() + "</query>");
+        } catch (SecurityException x) {
+        }

```

Comment:
in such case you increase Java warnings, could you add the comment inside catch?

Reply:
:D :D no problem, doing it now